File: src/sharing/service/file/SharingFileService.java
package sharing.service.file;

import java.util.List;

import sharing.entity.file.SharingFile;

public interface SharingFileService {
	public SharingFile findSharingFileById(Long sharingFileId) throws Exception;
	
	public Long addSharingFile(SharingFile sharingFile) throws Exception;

	public Long findAllFilesTotal() throws Exception;

	public List<SharingFile> findFilesByLimit(Long currentPage, Long pageSize) throws Exception;
}

File: WebContent/app/service/file/sharingFileService.js
define(['app'],function(app) {
	app.factory('SharingFileService',function($resource) {
		var sharingFileService = $resource('../SharingFileAction/:action', {});
		sharingFileService.addSharingFile = function(sharingFile,sucesscb,errorcb) {
			sharingFileService.save({action:"addSharingFile"},{sharingFile:sharingFile},sucesscb,errorcb);
		},
		sharingFileService.findAllFilesTotal = function(sucesscb,errorcb) {
			sharingFileService.get({action:"findAllFilesTotal"},sucesscb,errorcb);
		},
		sharingFileService.findFilesByLimit = function(currentPage,pageSize,sucesscb,errorcb) {
			sharingFileService.get({action:"findFilesByLimit",currentPage:currentPage,pageSize:pageSize},sucesscb,errorcb);
		},
		sharingFileService.findFilesAndTotalByLimit = function(currentPage,pageSize,sucesscb,errorcb) {
			sharingFileService.get({action:"findFilesAndTotalByLimit",currentPage:currentPage,pageSize:pageSize},sucesscb,errorcb);
		}
		return sharingFileService;
	})
})